flush-imports.js

```diff
@@ -0,0 +1,43 @@
+import { rollup } from "rollup";
+
+/**
+ * @param {Set<string>} resultSet
+ * @returns {Plugin}
+ */
+function rollupImportAnalyzerPlugin(resultSet) {
+  let buildOptions;
+  return {
+    name: "import-analyzer",
+    buildStart(options) {
+      buildOptions = options;
+    },
+    async resolveId(source, importer) {
+      const resolvedModule = await this.resolve(source, importer);
+
+      if (!buildOptions.input.includes(source)) {
+        resultSet.add({
+          importPath: source,
```

Comment:
If we could extract the location of the import in the AST, we could use that metainfo for future things, like the vscode extension, to highlight a potentially problematic imports in the file.